### PR TITLE
Support Creating/Reconfiguring a simulator VM with VApp properties

### PR DESCRIPTION
**Pull request by aruneshpa**
Currently, the simulator ignores any VApp properties in the VM
creation and Reconfigure workflows.

This change enables users to instantiate or Reconfigure a simulator VM
with VApp properties set. This commit also adds tests to validate the
workflows.

Closes: #2642

## Description

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

Closes: #(issue-number)

## Type of change

Please mark options that are relevant:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update
- [ ] Build related change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. If applicable, please also list any relevant
details for your test configuration.

- [x] Added unit tests to validate the change

## Checklist:

- [x] My code follows the CONTRIBUTION [guidelines](./../CONTRIBUTING.md) of
  this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/google/uuid v1.2.0
 	github.com/kr/pretty v0.1.0
 	github.com/kr/text v0.1.0 // indirect
+	github.com/stretchr/testify v1.7.0
```
Comment (dougm) [inline go.mod:13]: Let's not bring in the dep, govmomi tests just use the stdlib testing package.

File: go.sum [hunk 2]
```diff
@@ -1,13 +1,24 @@
 github.com/a8m/tree v0.0.0-20210115125333-10a5fd5b637d h1:4E8RufAN3UQ/weB6AnQ4y5miZCO0Yco8ZdGId41WuQs=
 github.com/a8m/tree v0.0.0-20210115125333-10a5fd5b637d/go.mod h1:FSdwKX97koS5efgm8WevNf7XS3PqtyFkKDDXrz778cg=
+github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
```
Comment (dougm) [inline go.sum:3]: Can you revert the go.sum changes too, assuming these are from testify ?

**Review by dougm — COMMENTED** [review]
Thanks @aruneshpa , looks good. Can you change the tests to avoid pulling in the testify package?

File: go.mod (hunk 1 above)
Comment (aruneshpa) [reply go.mod]: Thanks for the review, @dougm. I have removed the dependency and updated the test to use stdlib testing package.

File: go.sum (hunk 2 above)
Comment (aruneshpa) [reply go.sum]: Fixed.

**Review by dougm — APPROVED** [review]
Awesome, thanks @aruneshpa !

**Comment by embano1** [comment]
Hm, looks like the unit tests weren't passing. This breaks our pipeline. @aruneshpa can you please take a look and provide a quick fix?

**Pull request merged**